### ISSUE TITLE
Angular component are not getting destroyed

### ISSUE THREAD

**Issue by bytesofdhiren** [angular/angular#30724]
In angular application, every time I route to a new page, my earlier page's component is staying in memory. It's not getting destroyed and when I route back to that page it creates a new instance.

Application is **not created through angular CLI**. It's a custom build application using the webpack configuration provided on the angular website. Actually, I am doing performance improvement of the existing enterprise angular application and this demo app I created for isolation of issue.

Github [https://github.com/bytesofdhiren/angular-memory-issue](https://github.com/bytesofdhiren/angular-memory-issue)

Basically, every time I route to home or register page, the application creates a new instance of the register component and home component. It's not destroying the earlier created component.

Things I tried  
1. Lazy loading 
2. No route reuse strategy

Angular version: 5

![Screenshot from 2019-05-28 16:03:48](https://user-images.githubusercontent.com/15865266/58565797-fde0bc00-824c-11e9-9175-f44dba1e6b6c.png)

**Comment by alvaro450** [comment]
Try running the application in production mode, I have noticed these sort of issue only in development mode but not in production mode.

**Comment by bytesofdhiren** [comment]
That comes as a surprise and true. Closing the bug.
